pkg/virt-launcher/virtwrap/network/dhcp/filter.go

```diff
@@ -0,0 +1,68 @@
+// This file is inspired by github.com/krolaw/dhcp4/conn/filter.go.
+// Couldn't use the original one since SO_REUSEADDR option has to be set.
+package dhcp
+
+import (
+	"context"
+	"net"
+	"syscall"
+
+	"golang.org/x/net/ipv4"
+)
+
+// Creates listener on all interfaces and then filters packets not received by interfaceName
+func NewUDP4FilterListener(interfaceName, laddr string) (c *serveIfConn, e error) {
+	iface, err := net.InterfaceByName(interfaceName)
+	if err != nil {
+		return nil, err
+	}
+	lc := net.ListenConfig{
+		Control: func(network, address string, c syscall.RawConn) error {
+			var opErr error
+			err := c.Control(func(fd uintptr) {
+				opErr = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_REUSEADDR, 1)
+			})
+			if err != nil {
+				return err
+			}
+			return opErr
+		},
```

Comment:
Can you move this to a function with a proper name? Something like `setSOReuseaddr`.

Reply:
Extracted to `CreateListenConfig`.